File: 프로그래머스/2/42747. H－Index/H－Index.py
# 100,000,00 = 0.1초

def solution(citations):
    answer = 0
    citations.sort(reverse=True)    # 1000log1000
    for idx in range(len(citations)-1, -1, -1):   # 1000~1
        if citations[idx] >= idx+1:
            answer = idx+1
            return answer
    return 0    # [0, 0, 0]

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int n = citations.length;
        Integer[] books = new Integer[n];
        for (int i=0; i<n; i++) {
            books[i] = citations[i];
        }
        Arrays.sort(books, (a, b)->b-a);
        
        // 1000번 * 10000번
        int answer = -1;
        for (int hIndex = books[0]; hIndex >=0; hIndex--) {
            int highCount = 0;
            int lowCount = 0;
            for (int j=0; j<n; j++) {
                if (books[j] >= hIndex) highCount++;
            }
            if (highCount >= hIndex) {
                answer = hIndex;
                break;
            }
        }
        
        return answer;
    }
}